File: leetcode/levelOrder.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""
from collections import deque
from typing import List
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        rs, q = [], [root] if root else []
        while q:
            rs.append([node.val for node in q])
            # for node in q:
            #     for child in node.children:
            q = [child for node in q for child in node.children if child]
        return rs

File: leetcode/levelOrder.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  const rs = [];
  const dfs = (node, level = 0) => {
    if (!node) return;
    if (!rs[level]) rs[level] = [];
    rs[level].push(node.val);
    for (let c of node.children) {
      dfs(c, level + 1);
    }
  };
  dfs(root);
  return rs;
};

var levelOrder = function (root) {
  const rs = [];
  const stack = [[root, 0]];
  while (stack.length) {
    const [node, level] = stack.pop();
    if (!rs[level]) rs[level] = [];
    rs[level].push(node.val);
    for (let c of node.children) {
      stack.push([c, level + 1]);
    }
  }
  return rs;
};
